src/kdbindings/signal.h

```diff
@@ -352,6 +260,31 @@ class Signal
         return ConnectionHandle{ m_impl, m_impl->connect(slot) };
     }
 
+    /**
+     * @brief Establishes a deferred connection between the provided evaluator and slot.
+     *
+     * This function allows connecting an evaluator and a slot such that the slot's execution
+     * is deferred until the conditions evaluated by the `evaluator` are met.
+     *
+     * First argument to the function is reference to a shared pointer to the `ConnectionEvaluator` responsible for determining
+     * when the slot should be executed.
+     *
+     * @return An instance of ConnectionHandle, that can be used to disconnect
+     * or temporarily block the connection.
+     *
+     * @note
+     * The `KDBindings::Signal` class itself is not thread-safe. While the `ConnectionEvaluator` is inherently
+     * thread-safe, ensure that any concurrent access to this Signal is protected externally to maintain thread safety.
+     */
+    ConnectionHandle connectDeferred(const std::shared_ptr<ConnectionEvaluator> &evaluator, std::function<void(Args...)> const &slot)
```

Comment:
We probably do want that.
For this first PR, I would stick with the simple implementation though.

When I engineered the `::connect(Func &&slot, FuncArgs &&...args)`, I might have gotten a bit lost in the beautiful complexity of template meta programming and over-engineered this quite a bit :see_no_evil: .
Especially as any errors generated by this function are probably impossible to understand, I've ever since wondered whether it wouldn't have been better to produce a second overload for member functions, and leave everything else up to `std::bind`.
So I want to avoid adding something like this again without feedback.

@seanharmer @lemirep Do you have any insight into this? Is the current templated function usable? Would leaving this up to std::bind be a hassle at all?

Reply:
Sure. Let's not bog it down now.